sources/fedora-http.go

```diff
@@ -10,49 +11,90 @@ import (
 	"github.com/canonical/lxd-imagebuilder/shared"
 )
 
 type fedora struct {
-	common
+	commonRHEL
 }
 
 // Run downloads a container base image and unpacks it and its layers.
 func (s *fedora) Run() error {
-	baseURL := fmt.Sprintf("%s/packages/Fedora-Container-Base", s.definition.Source.URL)
+	// For backwards compatibility, fallback to manual URL construction
+	// when the release version is equal or less than 40.
+	relNum, err := strconv.Atoi(s.definition.Image.Release)
+	if err == nil && relNum <= 40 {
+		baseURL := fmt.Sprintf("%s/packages/Fedora-Container-Base", s.definition.Source.URL)
+
+		// Get latest build
+		build, err := s.getLatestBuild(baseURL, s.definition.Image.Release)
+		if err != nil {
+			return fmt.Errorf("Failed to get latest build: %w", err)
+		}
 
-	// Get latest build
-	build, err := s.getLatestBuild(baseURL, s.definition.Image.Release)
-	if err != nil {
-		return fmt.Errorf("Failed to get latest build: %w", err)
+		fname := fmt.Sprintf("Fedora-Container-Base-%s-%s.%s.tar.xz", s.definition.Image.Release, build, s.definition.Image.ArchitectureMapped)
+
+		// Download image
+		sourceURL := fmt.Sprintf("%s/%s/%s/images/%s", baseURL, s.definition.Image.Release, build, fname)
+
+		fpath, err := s.DownloadHash(s.definition.Image, sourceURL, "", nil)
+		if err != nil {
+			return fmt.Errorf("Failed to download %q: %w", sourceURL, err)
+		}
+
+		s.logger.WithField("file", filepath.Join(fpath, fname)).Info("Unpacking image")
+
+		// Unpack the base image
+		err = shared.Unpack(filepath.Join(fpath, fname), s.rootfsDir)
+		if err != nil {
+			return fmt.Errorf("Failed to unpack %q: %w", filepath.Join(fpath, fname), err)
+		}
+
+		s.logger.Info("Unpacking layers")
+
+		// Unpack the rest of the image (/bin, /sbin, /usr, etc.)
+		err = s.unpackLayers(s.rootfsDir)
+		if err != nil {
+			return fmt.Errorf("Failed to unpack: %w", err)
+		}
+
+		return nil
 	}
 
-	fname := fmt.Sprintf("Fedora-Container-Base-%s-%s.%s.tar.xz",
-		s.definition.Image.Release, build, s.definition.Image.ArchitectureMapped)
+	// Otherwise, get the ISO image URL from the available releases.
+	sourceURL, checksum, err := s.getDownloadURL(s.definition.Image.Release, s.definition.Image.Variant, s.definition.Image.ArchitectureMapped, ".iso")
+	if err != nil {
+		return err
+	}
 
-	// Download image
-	sourceURL := fmt.Sprintf("%s/%s/%s/images/%s", baseURL, s.definition.Image.Release, build, fname)
+	s.logger.Info("Downloading ISO image")
 
 	fpath, err := s.DownloadHash(s.definition.Image, sourceURL, "", nil)
 	if err != nil {
 		return fmt.Errorf("Failed to download %q: %w", sourceURL, err)
 	}
 
-	s.logger.WithField("file", filepath.Join(fpath, fname)).Info("Unpacking image")
+	fname := filepath.Join(fpath, filepath.Base(sourceURL))
 
-	// Unpack the base image
-	err = shared.Unpack(filepath.Join(fpath, fname), s.rootfsDir)
-	if err != nil {
-		return fmt.Errorf("Failed to unpack %q: %w", filepath.Join(fpath, fname), err)
+	// Verify checksum.
+	if checksum != "" {
```

Comment:
Why wouldnt a checksum be returned? This feels like it could introduce accidental issues in case the checksum is ever not returned we would not verify it?